lib/deref.js

```diff
@@ -247,23 +247,29 @@ module.exports = {
       }
       return { value: 'reference ' + schema.$ref + ' not found in the OpenAPI spec' };
     }
-    if (concreteUtils.compareTypes(schema.type, SCHEMA_TYPES.object) || schema.hasOwnProperty('properties') ||
-      schema.hasOwnProperty('additionalProperties')) {
+
+    const getAdditionalProperties = (schema) => {
+      let additionalProperties;
+      if (_.isBoolean(schema.additionalProperties)) {
+        additionalProperties = schema.additionalProperties;
+      }
+      else {
+        additionalProperties = this.resolveRefs(schema.additionalProperties, parameterSourceOption,
+          components, schemaResolutionCache, resolveFor, resolveTo, stack, _.cloneDeep(seenRef), stackLimit);
+      }
+      return additionalProperties;
+    };
+
+    if (concreteUtils.compareTypes(schema.type, SCHEMA_TYPES.object) || schema.hasOwnProperty('properties')) {
       // go through all props
       schema.type = SCHEMA_TYPES.object;
-      if (_.has(schema, 'properties') || _.has(schema, 'additionalProperties')) {
-        // shallow cloning schema object except properties object
+
+      if (_.has(schema, 'properties')) {
```

Comment:
@erfemega Again, it's not required that you always have `properties` present when defining `additionalProperties`. This will remove existing supported scenarios with just `additionalProperties` being defined.